### PR TITLE
Improved speed with autogenerated token check

### PR DESCRIPTION
**Pull request by kompotkot**
## Changes

Improved speed with autogenerated token check also security improvements for resources.

## How to test these changes?

Tested locally

### CODE REVIEW

**Comment by Andrei-Dolgolev** [comment]
My question why not in middleware with some flag wich allow or dissallow resticted tokens for mounted FastApi app?

File: brood/resources/api.py [hunk 1]
```diff
@@ -126,24 +128,38 @@ async def create_resource_handler(
...
+        request_user_authorization_with_groups
+    ),
     db_session=Depends(yield_db_read_only_session),
 ) -> data.ResourcesListResponse:
     """
     Get a list of available resources for the user.
 
     - **<query>** (string): Any query param to filter resources output by resource_data key
     """
+    is_token_restricted = user_authorization_with_groups[0]
```
Comment (zomglings) [inline brood/resources/api.py:141]: Just a small thing:

```
is_token_restricted, current_user_with_groups = user_authorization_with_groups
```

File: brood/resources/api.py [hunk 2]
```diff
@@ -166,14 +182,24 @@ async def get_resources_list_handler(
...
+        request_user_authorization
+    ),
     db_session=Depends(yield_db_session_from_env),
 ) -> data.ResourceResponse:
     """
     Provides resource information.
 
     - **resource_id** (uuid): Resource ID
     """
+    is_token_restricted = user_authorization[0]
```
Comment (zomglings) [inline brood/resources/api.py:195]: Suggest tuple unpacking as previous comment.

File: brood/resources/api.py [hunk 3]
```diff
@@ -211,6 +239,14 @@ async def update_resource_handler(
     - **update** (dict): Key-value pair to update
     - **drop_keys** (list): List of keys to drop
     """
+    is_token_restricted = user_authorization[0]
```
Comment (zomglings) [inline brood/resources/api.py:242]: Same

File: brood/resources/api.py [hunk 4]
```diff
@@ -241,14 +277,24 @@ async def update_resource_handler(
...
+        request_user_authorization
+    ),
     db_session=Depends(yield_db_session_from_env),
 ) -> data.ResourceResponse:
     """
     Delete existing resource.
 
     - **resource_id** (uuid): Resource ID
     """
+    is_token_restricted = user_authorization[0]
```
Comment (zomglings) [inline brood/resources/api.py:290]: Same

File: brood/resources/api.py [hunk 5]
```diff
@@ -291,6 +339,14 @@ async def add_resource_holder_permissions_handler(
     - **holder_type** (string): Type of holder (user or group)
     - **permissions** (list): List of permissions to add (admin, create, read, update, delete)
     """
+    is_token_restricted = user_authorization[0]
```
Comment (zomglings) [inline brood/resources/api.py:342]: Same

File: brood/resources/api.py [hunk 6]
```diff
@@ -337,6 +395,14 @@ async def get_resource_holders_permissions_handler(
     - **resource_id** (uuid): Resource ID
     - **holder_id** (uuid, null): User or group ID
     """
+    is_token_restricted = user_authorization[0]
```
Comment (zomglings) [inline brood/resources/api.py:398]: Same

File: brood/resources/api.py [hunk 7]
```diff
@@ -378,6 +446,14 @@ async def delete_resource_holder_permissions_handler(
     - **holder_type** (string): Type of holder (user or group)
     - **permissions** (list): List of permissions to add (admin, create, read, update, delete)
     """
+    is_token_restricted = user_authorization[0]
```
Comment (zomglings) [inline brood/resources/api.py:449]: Same

File: brood/resources/api.py [hunk 8]
```diff
@@ -109,6 +103,14 @@ async def create_resource_handler(
         - **application_id** (uuid)
         - **resource_data** (dict)
     """
+    is_token_restricted = user_authorization[0]
```
Comment (zomglings) [inline brood/resources/api.py:106]: Same

**Review by zomglings — COMMENTED** [review]
Small suggestion

Looks good to me, but I haven't tested.

**Pull request merged**